### PR TITLE
Added logging of .NET client HTTP requests

### PR DESCRIPTION
**Pull request by JamesNK**
https://github.com/aspnet/SignalR/issues/1391

### CODE REVIEW

File: src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LoggingHttpMessageHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Microsoft.AspNetCore.Sockets.Client.Http.Internal
+{
+    public class LoggingHttpMessageHandler : DelegatingHandler
+    {
+        private readonly ILogger<LoggingHttpMessageHandler> _logger;
+
+        public LoggingHttpMessageHandler(HttpMessageHandler inner, ILoggerFactory loggerFactory) : base(inner)
+        {
+            _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<LoggingHttpMessageHandler>();
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LoggingHttpMessageHandler.cs]: You don't need to be this defensive here. Just don't accept a null logger factory.

**Comment by davidfowl** [comment]
So we separately log web sockets?

**Comment by anurse** [comment]
Yeah, WebSockets don't use `HttpClient` (well technically the new implementation does but as an implementation detail) so we should also make sure we're logging the URL [there](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs#L327)

**Comment by davidfowl** [comment]
@JamesNK need some websocket love

File: src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs [hunk 2]
```diff
@@ -135,7 +139,10 @@ private HttpClient CreateHttpClient()
                 }
             }
 
-            var httpClient = httpMessageHandler == null ? new HttpClient() : new HttpClient(httpMessageHandler);
+            // Wrap message handler in a logging handler last to ensure it is always present
+            httpMessageHandler = new LoggingHttpMessageHandler(httpMessageHandler, _loggerFactory);
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs]: Might be worth checking if `_loggerFactory` is a `NullLoggerFactory`. Not necessary for this PR though

File: src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LoggingHttpMessageHandler.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LoggingHttpMessageHandler.cs]: copyright

**Pull request merged**